### PR TITLE
feat: show events on profile detail page

### PR DESCRIPTION
**Pull request by nac62116**
## List issues that are affected

see #288

### CODE REVIEW

File: app/routes/profile/$username/index.tsx [hunk 1]
```diff
@@ -145,15 +118,42 @@ export const loader: LoaderFunction = async (
...
+  // TODO: Type issues, rework public fields
+  let data: Partial<
+    NonNullable<Awaited<ReturnType<typeof getProfileByUsername>>>
+  > = {};
   for (const key in profile) {
+    // Only show public fields if user is anon, show all fields if user is not anon
     if (mode !== "anon" || publicFields.includes(key)) {
-      // @ts-ignore <-- Partials allow undefined, Profile not
-      data[key] = profile[key];
+      // Event relations must be transformed
```
Comment (manuelportela) [inline app/routes/profile/$username/index.tsx:138]: use functions instead of comments (precise naming)

**Review by manuelportela — CHANGES_REQUESTED** [review]
see comments

File: app/routes/profile/$username/index.tsx (hunk 1 above)
Comment (nac62116) [reply app/routes/profile/$username/index.tsx:138]: Created a ticket:
see #379

File: app/lib/event/utils.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        // @ts-ignore
+        rootEvents.push(rootItem);
+      }
+    })
+  );
+
+  return rootEvents;
+}
+
+export function getPublishedAndSortedEvents(
```
[the remark this answers is not in the document]
Comment (nac62116) [reply app/lib/event/utils.ts]: Resolved in 165e011

File: app/routes/profile/$username/index.tsx (hunk 1 above)
Comment (nac62116) [reply app/routes/profile/$username/index.tsx]: Resolved in 303171c

**Review by manuelportela — APPROVED** [review]

**Pull request merged**